src/java/org/apache/cassandra/config/DatabaseDescriptor.java

```diff
@@ -168,6 +168,41 @@
                                        ? new CommitLogSegmentManagerCDC(c, DatabaseDescriptor.getCommitLogLocation())
                                        : new CommitLogSegmentManagerStandard(c, DatabaseDescriptor.getCommitLogLocation());
 
+    static void updateInitialized(boolean clientInitialized, boolean toolInitialized, boolean daemonInitialized)
+    {
+        DatabaseDescriptor.clientInitialized = clientInitialized;
+        DatabaseDescriptor.toolInitialized = toolInitialized;
+        DatabaseDescriptor.daemonInitialized = daemonInitialized;
+    }
+
+    /**
+     * This method can be called from tests if we need to re-initialize the DD
+     * again with a new config, partitioner or snitch. Note that this method will
+     * not delete directories or null the config, it simply resets the variables
+     * that prevent changing config or directories.
+     * <p/>
+     * It cannot set the config to null because this would cause NPEs in the
+     * schedulers that are still running, e.g. TPC.
+     * <p/>
+     * It cannot delete directories because on remote storage this would result
+     * in errors if the test containers for remote storage are being shutdown
+     * concurrently. The caller should delete any directories if required.
```

Comment:
how is the caller supposed to know if it is required to remove some specific directories? is there a list somewhere?

Reply:
I don't think so. I guess this is meant as advice to anyone wanting to use this method in a test somewhere and is considering pushing deletion of directories here from those tests.

Perhaps that is also the intent of the previous text about setting config to null? Rather, more of a warning to someone later thinking it might be an easy way to reset some setting for a test scenario.

Reply:
Would it be better if the wording was `It cannot be used to delete directories...`?

And `It cannot be used to set the config to null...` - although it still seems to me that paragraph could be removed since the "initialized" fields are reset, it seems the intent is that the config should be reloaded again during one of the initialization methods.

Reply:
regardless of wording, after reading that comment I essentially know that if I use `resetUnsafe` I'm breaking something (removal of some directories), and have no idea how to fix it. So I either spend time figuring out (unlikely) or just :shrug:

what about adding a comment asking whoever runs into issues with not deleted dirs to expand this comment with what they learn? :)

Reply:
> if I use resetUnsafe I'm breaking something (removal of some directories)

To me, that's not what it's saying; it's saying `resetUnsafe` does not remove any directories and that is because it would be unsafe (haha, given context) to delete them and instead advises the caller to delete directories if the caller needs that. In other words, if you need directories deleted (presumably you know why and know the context) or run into problems with dirs that aren't deleted, handle it in your code, don't put it here.

I poked around some of the cndb tests that use this, and all of them call `ConfigurationControl.resetUnsafe` which does delete some directories. But I don't know any more either 🤷🏻